### ISSUE TITLE
'prop-types' rule doesn't recognize 'this-format' properties name

### ISSUE THREAD

**Issue by AlexKVal** [yannickcr/eslint-plugin-react#118]
I wasn't been able to find out such issue. Sorry if I missed one.

```js
propTypes: {
  ...
  'aria-controls': React.PropTypes.string
},

let {
  ...
  'aria-controls': ariaControls,  // <- error 'undefined' is missing in props validation  react/prop-types
  ...props } = this.props; // eslint-disable-line object-shorthand
```
```
error  'undefined' is missing in props validation  react/prop-types
```

**Comment by mathieumg** [comment]
I imagine this could be causing problem with that: https://github.com/yannickcr/eslint-plugin-react/pull/112/files#diff-531a13f7edd3ed4e7c65c3a7df7052f0R153

**Comment by AlexKVal** [comment]
I am sorry, I forgot to explicitly note that I'm checking it with the last versions:
`eslint@0.23.0`
`eslint-plugin-react@2.5.2`

I've checked now it against [the latest at this moment commit](https://github.com/yannickcr/eslint-plugin-react/tree/c0ed5fc3d7145d111b619b3fea7507b464a47fb6)
It throws the next error:
(exactly because of that line `'aria-controls': ariaControls,`,
I've checked it by commenting it out)
```
react-bootstrap/ on pr/857 
› npm run lint                

> react-bootstrap@0.23.4 lint /Users/alexkval/src/js/react-bootstrap
> eslint ./

/Users/alexkval/src/js/eslint-plugin-react/lib/rules/prop-types.js:436
        isIgnored(name.split('.').pop()) ||
                      ^
TypeError: Cannot read property 'split' of undefined
    at reportUndeclaredPropTypes (/Users/alexkval/src/js/eslint-plugin-react/lib/rules/prop-types.js:436:23)
    at EventEmitter.Program:exit (/Users/alexkval/src/js/eslint-plugin-react/lib/rules/prop-types.js:521:9)
    at EventEmitter.emit (events.js:129:20)
    at Controller.controller.traverse.leave (/Users/alexkval/src/js/react-bootstrap/node_modules/eslint/lib/eslint.js:736:25)
    at Controller.__execute (/Users/alexkval/src/js/react-bootstrap/node_modules/eslint/node_modules/estraverse/estraverse.js:393:31)
    at Controller.traverse (/Users/alexkval/src/js/react-bootstrap/node_modules/eslint/node_modules/estraverse/estraverse.js:481:28)
    at EventEmitter.module.exports.api.verify (/Users/alexkval/src/js/react-bootstrap/node_modules/eslint/lib/eslint.js:721:24)
    at processText (/Users/alexkval/src/js/react-bootstrap/node_modules/eslint/lib/cli-engine.js:201:27)
    at processFile (/Users/alexkval/src/js/react-bootstrap/node_modules/eslint/lib/cli-engine.js:226:12)
    at /Users/alexkval/src/js/react-bootstrap/node_modules/eslint/lib/cli-engine.js:322:26
```

It seems @Cellule would be interested to take a look into it :wink:

**Comment by Cellule** [comment]
I would like to check into it indeed :), but maybe not today, I'm pretty busy

**Comment by Cellule** [comment]
Could you share the exact test.
I did check the code you shared and I found a bug in `jsx-sort-prop-types` not handling `'this-format'` #123.
However, from what I see, `prop-types` does handle that case. I'm curious to see what's causing that crash.

**Comment by Cellule** [comment]
Never mind, I got it. I will create a PR for it very shortly.
